gpflow/utilities/utilities.py

```diff
@@ -39,6 +41,20 @@ def to_default_float(x):
     return tf.cast(x, dtype=default_float())
 
 
+class ShapeError(AssertionError):
+    pass
+
+
+def assert_shapes(shapes, *args, **kwargs):
```

Comment:
Is this custom `assert_shapes` really necessary? I, personally, like to stay as close to raw tensorflow as much as possible.

Reply:
@joelberkeley-pio pointed out that ValueErrors are commonly try:except:ed and that this would hide the shape asserts. I think ideally TensorFlow's `assert_shapes` would raise a custom exception, not a ValueError; this wrapper makes clearer what went wrong. I think it'll make it a bit more user-friendly.

Can take it out again if that's the majority preference...